### PR TITLE
duckstation: 0.pre+date=2022-01-18 -> unstable-2022-08-22

### PR DESCRIPTION
**Pull request by SuperSamus**
###### Description of changes

Updates DuckStation to the latest version.
There have been quite a few changes, for example the nogui interface is no longer built by default, and the project migrated to Qt6.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/emulators/duckstation/default.nix [hunk 1]
```diff
@@ -42,12 +44,11 @@ mkDerivation rec {
   buildInputs = [
     SDL2
     curl
-    gtk3
-    libevdev
     libpulseaudio
+    libXrandr
     mesa
     qtbase
-    sndio
```
Comment (SuperSamus) [inline pkgs/applications/emulators/duckstation/default.nix:50]: sndio is technically used by Cubeb (tier 2 support (out of 4)), I don't know how important it is, though.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -1431,7 +1431,7 @@ with pkgs;
 
   dosbox-staging = callPackage ../applications/emulators/dosbox-staging { };
 
-  duckstation = libsForQt5.callPackage ../applications/emulators/duckstation {};
+  duckstation = qt6Packages.callPackage ../applications/emulators/duckstation {};
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:1434]: Remember me being more explicit next time.

I am trying to get rid of `rec` in my expressions, however many expressions that use `qt.*Packages` fail miserably.

**Review by AndersonTorres — APPROVED** [review]
Remember me merge this Friday.

**Comment by SuperSamus** [comment]
It's Friday.
Though, thinking about it, this version is labelled as a "Preview Build", so maybe it'd be better to wait for the next Development build instead, or alternatively use the current one (Jul 08).

**Comment by AndersonTorres** [comment]
OK. I will add another week - remind me again next Friday.

**Comment by SuperSamus** [comment]
I left it as draft because I didn't want it to be merged.

> This version is labelled as a "Preview Build", so maybe it'd be better to wait for the next stable build instead, or alternatively use the current one (Jul 08).

I'll make a PR to revert it to the current stable.

**Pull request merged**